Mathlib/Data/Vector/Basic.lean

```diff
@@ -511,9 +511,9 @@ def casesOn₃ {motive : ∀ {n}, Vector α n → Vector β n → Vector γ n 
     motive v₁ v₂ v₃ :=
   inductionOn₃ (C := motive) v₁ v₂ v₃ nil @fun _ x y z xs ys zs _ => cons x y z xs ys zs
 
-/-- Cast a vector to an array. -/
-def toArray : Vector α n → Array α
-  | ⟨xs, _⟩ => cast (by rfl) xs.toArray
+/-- Convert a `Vector` to an `Array`. -/
+@[deprecated toVector (since := "20/05/2025")]
```

Comment:
Why the deprecation? This isn't even an equivalent function.

Reply:
Oh, that is leftover from when we were going to deprecate it, I think? I think my argument was that it is more correct to convert it to a Vector and then to an array if you need.

Reply:
If you'd rather I just take the deprecation out, I can.

Reply:
I'm generally in favor of cutting out `def`s for combinations of projections (and so in favor of this deprecation), because if you keep them then you need to write API for them; but I'll let @digama0 make the call here.